src/GPUActivityAgent.cpp

```diff
@@ -314,6 +322,23 @@ namespace geopm
             f_request = std::min(f_request, m_resolved_f_gpu_max);
             f_request = std::max(f_request, m_resolved_f_gpu_efficient);
 
+            if (phi >= 0.5) {
+                if (!std::isnan(gpu_utilization) &&
+                    gpu_utilization == 0) {
+                    if (m_gpu_idle_timer.at(domain_idx) > 0) {
+                        m_gpu_idle_timer.at(domain_idx) = m_gpu_idle_timer.at(domain_idx) - 1;
+                    }
+                }
+                else {
+                    m_gpu_idle_timer.at(domain_idx) = 10;
```

Comment:
The width of the idle filter should probably be calculated or at least #defined close to the agent's wait time in case wait time becomes variable in the future

Reply:
That comment assumes this is supposed to approximate a time-based filter. If it is actually meant to be 10 samples regardless of time, then also add a comment explaining why.

Reply:
Updated to a define of 10 samples regardless of time requested, with a comment explaining why.